src/test/kotlin/com/pubnub/api/subscribe/eventengine/effect/HandshakeEffectTest.kt

```diff
@@ -4,46 +4,74 @@ import com.pubnub.api.PubNubException
 import com.pubnub.api.endpoints.remoteaction.RemoteAction
 import com.pubnub.api.enums.PNOperationType
 import com.pubnub.api.enums.PNStatusCategory
+import com.pubnub.api.eventengine.EventSink
 import com.pubnub.api.models.consumer.PNStatus
-import org.junit.jupiter.api.Assertions.assertTrue
+import com.pubnub.api.subscribe.eventengine.event.Event
+import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
+import io.mockk.spyk
+import io.mockk.verify
+import org.awaitility.Awaitility
+import org.awaitility.Durations
 import org.junit.jupiter.api.Test
-import java.util.concurrent.CountDownLatch
+import java.time.Duration
 import java.util.concurrent.Executors
-import java.util.concurrent.TimeUnit
 
-class RemoteActionManagedEffectTest {
+class HandshakeEffectTest {
+    private val eventSink = TestEventSink()
+    private val subscriptionCursor = SubscriptionCursor(1337L, "1337")
+    private val reason = PubNubException("Unknown error")
+
     @Test
-    fun `failing handshake runs the onCompletion block`() {
+    fun `should deliver HandshakeSuccess event when HandshakeEffect succeeded`() {
         // given
-        val latch = CountDownLatch(2)
-        val failingManagedEffect = failingRemoteAction<Int>().toManagedEffect { _, s ->
-            if (s.error) {
-                latch.countDown()
-            }
-        }
+        val handshakeEffect = HandshakeEffect(successfulRemoteAction(subscriptionCursor), eventSink)
 
         // when
-        failingManagedEffect.runEffect { latch.countDown() }
+        handshakeEffect.runEffect()
 
         // then
-        assertTrue(latch.await(100, TimeUnit.MILLISECONDS))
+        Awaitility.await()
+            .atMost(Durations.ONE_SECOND)
+            .with()
+            .pollInterval(Duration.ofMillis(20))
+            .untilAsserted { listOf(Event.HandshakeSuccess(subscriptionCursor)) == eventSink.events }
```

Comment:
But `untilAsserted` need an assertion of some sort inside. It can't be just a boolean that is being removed.